windows/vhba_hot_add_remove/hotremove_disk_and_ctrl.yml

```diff
@@ -0,0 +1,87 @@
+# Copyright 2021-2024 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+# Hot remove the disks added to the new disk controller firstly,
+# then remove that controller when disks are removed successfully,
+# or there will be error with below message when removing NVMe controller:
+# "The guest operating system did not respond to a hot-remove request
+# for device nvme0 in a timely manner."
+#
+- name: "Get disk and controller number before hot removing disk"
+  include_tasks: get_guest_disk_ctrl_num.yml
+  vars:
+    disk_ops_stage: 'before'
+
+- name: "Get VM disk controller info before hot removing disk"
+  include_tasks: ../../common/vm_get_disk_controller_facts.yml
+
+- name: "Set fact of the number of disks attached to the controller"
+  ansible.builtin.set_fact:
+    disk_num_under_remove: >-
+      {{ disk_controller_facts['disk_controller_data'][disk_controller][new_vhba_bus_number | string]['controller_disks_devicekey'] | default([]) }}
+
+- name: "Print the number of the disks attached to the controller"
+  ansible.builtin.debug:
+    msg: "Disks attached to the '{{ test_disk_ctrl_type }}' controller will be removed: {{ disk_num_under_remove | length }}"
+
+- name: "Hot remove disks attached to the controller"
+  when: disk_num_under_remove | length > 0
+  block:
+    - name: "Hot remove disk"
+      include_tasks: ../../common/vm_hot_add_remove_disk.yml
+      vars:
+        disk_operation: 'absent'
+        ctrl_number: "{{ new_vhba_bus_number }}"
+        unit_number: "{{ rm_disk_unit_num }}"
+        disk_controller_type: "{{ test_disk_ctrl_type }}"
+      loop: "{{ range(0, (disk_num_under_remove | length)) }}"
+      loop_control:
+        loop_var: rm_disk_unit_num
+
+    - name: "Wait 10 seconds after disk hot removing"
+      ansible.builtin.pause:
+        seconds: 10
+
+    - name: "Get disk number in guest OS after hot removing"
+      include_tasks: ../utils/win_get_disk_num.yml
+
+    - name: "Handle known issue on NVMe disk hot remove"
+      include_tasks: handle_nvme_ops_known_issue.yml
+      when:
+        - disk_num_guest | int != disk_num_guest_before | int - disk_num_under_remove | length
```

Comment:
The value of disk_num_guest is changed in `handle_nvme_ops_known_issue.yml`, it will affect the tasks after `disk_num_guest` setting. I suggest adding `disk_num_guest_after` before this task too, and use `disk_num_guest_after` in this when and next assert task.

Reply:
Yeah, thanks for catching this.